### PR TITLE
Edge: Send out LocationEvents for #fragment navigation

### LINKED ISSUE #1581: SWT Browser component (Windows Edge): URLs sent to LocationListener are not complete
**Describe the bug**
In Eclipse APP4MC we are using SVG diagrams to show specific aspects of the data model.
The diagram is:
 - generated by PlantUML
 - displayed in SWT Browser component
 - contains links like **href="#obj37"**

The links are used to navigate in the model and select the corresponding object in the model editor (handled by a LocationListener).

The LocationListener of 
 -  **IE browser component** can be user to handle the clicks by analyzing the URL
 -  **Edge browser component** is not receiving the complete URL

**Details**
- IE links look like: `about:blank#obj37`
 - Edge links look like: `file:///C:/Users/abc/AppData/Local/Temp/base12703840119113867392.html#obj37`
 - Browser is setup with LocationListener like this:
    ```
   browser.addLocationListener(LocationListener.changingAdapter(c -> {
        int idx = c.location.lastIndexOf('#');
        if (idx >= 0) {
            <compute link to object>
            <sent (topic + link) to IEventBroker to select object in editor>
        }
    }));
    ```

**Screenshots**
SVG_displayed_in_SWT_Browser

**Environment:**
1. Select the platform(s) on which the behavior is seen:
- - [ ] All OS
- - [x] Windows
- - [ ] Linux
- - [ ] macOS 

2. Additional OS info (e.g. OS version, Linux Desktop, etc)
Windows 11
 
**Version**
 Eclipse 2024-12 M2

### PR DESCRIPTION
**Pull request by sratz**
Resolves #1581.

### CODE REVIEW

**Comment by sratz** [comment]
Any objection to merge this for RC1?

It fixes a regression introduced with #1451.

The fix was validated in the associated issue.

@HeikoKlare could you have a look at the code changes?

@akurtakov @merks Pinging you for PMC approval.

**Comment by merks** [comment]
It's quite a bit of code that I will not have time review right now.

If you and @HeikoKlare agree it's a good thing then I'm happy to defer to your good judgement:

PMC +1

File: bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java
Comment (HeikoKlare) [inline bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java:994]: This seems to be mostly duplicate code to what happens in `handleDocumentTitleChanged()`. Can that be extracted into a separate, parameterized method?

**Review by HeikoKlare — APPROVED** [review]
The change looks sound and since it has been validated in the issue thread and only affects Edge (which is an opt-in feature), I consider this safe to merge for RC1.

I only see a potential code duplication that we might clean up before merging. But I leave it up to you whether to improve that or not. You have my approval to merge this in either case.

File: bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java
Comment (sratz) [reply bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java:994]: Yes, indeed. There's several places in Edge class that could be simplified.

I think we should have a round of refactoring and deduplication beginning of next dev cycle.

I'll merge this as-is for now.

**Pull request merged**
